plugins/woocommerce/src/Internal/StockNotifications/Admin/NotificationsPage.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+declare( strict_types = 1 );
+
+namespace Automattic\WooCommerce\Internal\StockNotifications\Admin;
+
+use Automattic\WooCommerce\Internal\StockNotifications\Admin\NotificationsListTable;
+use Automattic\WooCommerce\Internal\StockNotifications\Notification;
+
+/**
+ * Notifications admin page for Customer Stock Notifications.
+ */
+class NotificationsPage {
+
+	/**
+	 * Page URL.
+	 *
+	 * @const PAGE_URL
+	 */
+	const PAGE_URL = 'admin.php?page=wc-customer-stock-notifications';
+
+	/**
+	 * Init.
+	 *
+	 * @internal
+	 *
+	 * @return void
+	 */
+	final public function init() {
+		// Select section.
+		$section = '';
+
+		// Nonce is checked in NotificationsPage::delete and NotificationsPage::output just displays the page.
+		if ( isset( $_GET['section'] ) ) { // phpcs:ignore WordPress.Security.NonceVerification.Recommended
+			$section = wc_clean( wp_unslash( $_GET['section'] ) ); // phpcs:ignore WordPress.Security.NonceVerification.Recommended
+		}
+
+		switch ( $section ) {
+			case 'delete':
+				$this->delete();
+				break;
+			default:
+				$this->output();
+				break;
+		}
+	}
+
+	/**
+	 * Render page.
+	 */
+	public static function output() {
+
+		if ( isset( $_GET['notice'] ) ) { // phpcs:ignore WordPress.Security.NonceVerification.Recommended
```

Comment:
Does this mean that we will be left with a notice GET in the URL? This suggests that every refresh will produce the same notice regardless of the action, as long as the GET is in the URL. Can we explore a way to avoid leaving footprints in the URL after actions?